client/state/sites/hooks/use-site-global-styles-status.ts

```diff
@@ -1,23 +1,58 @@
 import { useQuery } from '@tanstack/react-query';
 import wpcom from 'calypso/lib/wp';
+import { useSelector } from 'calypso/state';
+import { getSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
 export type GlobalStylesStatus = {
 	shouldLimitGlobalStyles: boolean;
 	globalStylesInUse: boolean;
+	globalStylesInPersonalPlan: boolean;
 };
 
 // While we are loading the Global Styles Info we can't assume that we should limit global styles, or we would be
 // showing notices for paid sites until we fetch the data from the server.
-export const DEFAULT_GLOBAL_STYLES_INFO: GlobalStylesStatus = {
+const DEFAULT_GLOBAL_STYLES_INFO: GlobalStylesStatus = {
 	shouldLimitGlobalStyles: false,
 	globalStylesInUse: false,
+	globalStylesInPersonalPlan: false,
 };
 
-export const getGlobalStylesInfoForSite = ( siteId: number | null ): GlobalStylesStatus => {
+function isObject( x: unknown ): x is Record< string, unknown > {
+	return typeof x === 'object' && x !== null;
+}
+
+const getExperimentAssignment = ( experimentName: string ): string | null => {
+	return wpcom.req
+		.get(
+			{
+				path: '/experiments/0.1.0/assignments/calypso',
+				apiNamespace: 'wpcom/v2/',
+			},
+			{
+				experiment_name: experimentName,
+			}
+		)
```

Comment:
I don't think we should be pinging the explatEndpoint directly when we already have the necessary helpers defined.

https://github.com/Automattic/wp-calypso/blob/751c0cfef75805287aafd55672f2cd2af7c730d4/client/lib/explat/README.md?plain=1#L1-L17

CC - @aaronyan @jessie-ross

Reply:
The reason for this is that those helpers don't work on SSR rendered pages. There's a discussion about it here: paYJgx-3yE-p2#comment-3668

Reply:
See my reply here, please do not do this:
https://trexp2.wordpress.com/2023/06/30/global-styles-on-personal-a-b-gating-logic/#comment-3703